platform-api/src/main/java/io/datacater/core/stream/KafkaStreamsAdmin.java

```diff
@@ -309,7 +329,22 @@ public StreamSpec createStream(StreamSpec spec) {
     if (!streamExists()) {
       NewTopic newTopic = new NewTopic(this.name, this.partitions, this.replication);
       newTopic.configs(spec.getConfig());
-      admin.createTopics(List.of(newTopic));
+      KafkaFuture<Config> createTopicConfigFuture =
+          admin.createTopics(List.of(newTopic)).config(this.name);
```

Comment:
As far as i can tell, we only call `.config(this.name)` so we receive an object wrapped in `KafkaFuture` to check if the topic could be created.
if we change this line to:
`KafkaFuture<Void> createTopicConfigFuture = admin.createTopics(List.of(newTopic)).all();`
The metadata and configs aren't mapped to the response and we can still use redpanda

Reply:
That's a great idea. We currently do not check the returned metadata, so happy to implement the suggested change.